packages/mongodb-dialects/mongosh/src/main/kotlin/com/mongodb/jbplugin/dialects/mongosh/query/References.kt

```diff
@@ -29,9 +29,9 @@ fun <S> MongoshBackend.resolveValueReference(
 
 fun <S> MongoshBackend.resolveFieldReference(fieldRef: HasFieldReference<S>) =
     when (val ref = fieldRef.reference) {
-        is FromSchema -> registerConstant(ref.fieldName)
-        is Inferred<S> -> registerConstant(ref.fieldName)
-        is HasFieldReference.Computed<S> -> registerConstant(ref.fieldName)
+        is FromSchema -> registerConstant(ref.displayName)
```

Comment:
Changed to use displayName because fieldName removes any prefixed '$' in the field, and in unwind it's mandatory. Because displayName holds the information "as the user wrote it" makes sense to use it here. We were using fieldName because this part was done before displayName was introduced.